lib/MongoDB/Database.pm

```diff
@@ -232,15 +232,16 @@ L</list_collections> to iterate over collections instead.
 =cut
 
 sub collection_names {
-    my ( $self, $filter ) = @_;
+    my ( $self, $filter, $options ) = @_;
     $filter ||= {};
+    $options ||= {};
```

Comment:
As described in `rename`, we should pull out and validate/get implicit session here and then pass explicitly into the op.